asciimatics/screen.py

```diff
@@ -1914,6 +1914,7 @@ def close(self, restore=True):
                 curses.echo()
                 curses.nocbreak()
                 curses.endwin()
+                signal.signal(signal.SIGWINCH, signal.SIG_DFL)
```

Comment:
To be honest, I don't see why we don't just reset everything every time on curses.  We can't do that on Windows, but I think the restore logic is pointless on curses.  I've done a quick test and I can run all the tests and an arbitrary selection of samples without any errors/failures if I remove the `if restore:` line.

All in all, I think it's best to reset everything properly - i.e. remove the "if" statement and simply close down curses and reinstate any old signal handlers whenever `close` is called.